test/blackbox-tests/test-cases/promote/deep-subdir.t/run.t

```diff
@@ -0,0 +1,43 @@
+Tests for promoting directory targets
```

Comment:
Generally we try to avoid cram test folders with multiple files. Can you rename this file into `deep-subdir.t` and inline the `dune` and `dune-project`?

Otherwise the test looks good to me!

Reply:
Thanks! And sorry for the delay: I missed the notification...

I now see that @ElectreAAS has [incorporated](https://github.com/ocaml/dune/pull/11226/commits/ea5bbbf1de50a0054a7150643fada86f158bd482) this PR into #11226, with your suggestion and along with a fix (thanks a lot!). Should I close this one then?

Reply:
As you want, I incorporated your test because I wanted to be sure I wasn't making any existing bug worse, but that doesn't mean you have to close this PR
If you can apply Marek's suggestion we can merge this and I'll modify #11226 accordingly

Reply:
Done. (And signed the commits.)